### ISSUE TITLE
Let guests choose their own name

### ISSUE THREAD

**Issue by dampfklon** [hackmdio/hackmd#268]
As the title states let unregistered people choose their shown nickname, like its possible in etherpad-lite

**Comment by jackycute** [comment]
Hi @dampfklon,
Currently we are NOT saving unregistered people (guest) authorship, so I think it might not needed be able to change guest name.
BTW, hackpad also can't change guest name.

**Comment by dampfklon** [comment]
Hi
its not about saving the authorship, the current cursor positions are currently shown in the editor with the generic guest names, those are not really usefull so I thought changing them to a meaningfull name would be nice

**Comment by jackycute** [comment]
Yeah, now we'll give a random name from a package called `chance`.
This sounds like a reasonable idea but it's only last for a session.

**Comment by dampfklon** [comment]
I can only argue from our use case. We use hackmd to write protocols of online meetings while those take place many people simultaneously work on a document. In this case the generic names are bad because you can not see who writes. After the meeting is over the document no longer is edited. So a name only last a session would no problem.

**Comment by jackycute** [comment]
OK, let's do this.

**Comment by ccoenen** [comment]
This also extends to "let people in general choose their own name". Right now, I'm registered with an email address starting with `mail@<something>`. My Hackmd calls me `mail` - I haven't found a place to change it, yet.

**Comment by jackycute** [comment]
@ccoenen That's another issue #314

**Comment by SISheogorath** [comment]
Some hints:

https://github.com/hackmdio/hackmd/blob/ea9b1557f9834b2b140bd9b07488b5fca3953791/lib/realtime.js#L595-L623

Here we generate the username. Currently, there is no way to update it afterward. It's not a trivial issue, but can be done by someone who is familiar to JS.

**Comment by dakira** [comment]
This would be great. We have the exact same usecase as @dampfklon (+ the same problem with many users just called "mail" because that's how their email address begins).
